trinity/plugins/builtin/ethstats/ethstats_service.py

```diff
@@ -134,13 +134,19 @@ def get_node_block(self) -> EthstatsData:
         }
 
     async def get_node_stats(self) -> EthstatsData:
-        response: PeerCountResponse = await self.context.event_bus.request(
-            PeerCountRequest()
-        )
+
+        try:
+            peer_count = (await asyncio.wait_for(
+                self.context.event_bus.request(PeerCountRequest()),
+                timeout=0.5
+            )).peer_count
+        except asyncio.TimeoutError:
+            self.logger.warning("Timeout: PeerPool did not answer PeerCountRequest")
+            peer_count = 0
```

Comment:
Maybe instead of defaulting to zero it should retry up to `N` times or for up to `T` seconds since this is more likely a race condition that if we wait a little we should be able to get a *real* answer.

I don't feel strongly about this.

Reply:
That's true. I deliberately went for a stop gap solution here because I believe we want to revisit this anyway when we start adding more infos to the `get_node_stats` response. So if you don't feel strongly about getting this fixed *now* I'd prefer to just keep it as is for now.